test/query_test.exs

```diff
@@ -65,6 +65,22 @@ defmodule QueryTest do
     assert [[[[0]]]] = query("SELECT ARRAY[ARRAY[0]]", [])
   end
 
+  test "decode array domain", context do
+    assert [[[1.0, 2.0, 3.0]]] =
+           query("SELECT ARRAY[1, 2, 3]::floats_domain", [])
+
+    assert [[[%Postgrex.Point{x: 1.0, y: 1.0}, %Postgrex.Point{x: 2.0, y: 2.0}, %Postgrex.Point{x: 3.0, y: 3.0}]]] =
+           query("SELECT ARRAY[point '1,1', point '2,2', point '3,3']::points_domain", [])
+  end
+
+  test "encode array domain", context do
+    floats = [1.0, 2.0, 3.0]
+    assert [[^floats]] = query("SELECT $1::floats_domain", [floats])
+
+    points = [%Postgrex.Point{x: 1.0, y: 1.0}, %Postgrex.Point{x: 2.0, y: 2.0}, %Postgrex.Point{x: 3.0, y: 3.0}]
+    assert [[^points]] = query("SELECT $1::points_domain", [points])
```

Comment:
Can we cast the result to text so that we encode the parameter to binary but decode the result as text in this test, i.e. `SELECT $1::points_domain::text`.

Reply:
The reason to do this is so that we have encode tests that don't depend on our decoding. It is okay to have an identity test too but not required.